### PR TITLE
ChartProperties refactor and Y axis time format

### PR DESCRIPTION
**Pull request by sglogovic**
- Refactored ChartProperties to make it easier to add new properties
- Added Y axis category and format to enable Time as axis value.

### CODE REVIEW

File: Acceleratio.XCellKit/ChartExport/Chart/ChartProperties/BaseChartProperties.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        public virtual bool AxisX { get; set; } = true;
+
+        /// <summary>
+        /// Set if Y Axis is visible
+        /// </summary>
+        public virtual bool AxisY { get; set; } = true;
+
+        public virtual string AxisYFormatCategory { get; set; } = "General";
+
+        public virtual string AxisYFormatCode { get; set; } = "General";
```
Comment (vinkobedek) [inline Acceleratio.XCellKit/ChartExport/Chart/ChartProperties/BaseChartProperties.cs:54]: FormatCategory, FormatCode, what values are possible?

Comment (sglogovic) [reply Acceleratio.XCellKit/ChartExport/Chart/ChartProperties/BaseChartProperties.cs:54]: FormatCode can be basically anything... For example [h]:mm;@, $#,##0.00, etc...
AxisYFormatCategory can be General ,Number, Currency and rest, mostly just so Chart knows what type of data can be expected

File: Acceleratio.XCellKit/ChartExport/Chart/ChartPropertiesSetup.cs [hunk 2]
```diff
@@ -3,58 +3,26 @@
...
-        /// Set color for each series.
-        /// </summary>
-        public virtual List<string> SeriesColor { get; set; }
+        private bool isArgumentDate { get; set; } = false;
 
-        /// <summary>
-        /// Set if Legend is visible
-        /// </summary>
-        public virtual bool Legend { get; set; } = true;
+        private int dataCount { get; set; } = 0;
```
Comment (vinkobedek) [inline Acceleratio.XCellKit/ChartExport/Chart/ChartPropertiesSetup.cs:22]: _name for privates?

Comment (sglogovic) [reply Acceleratio.XCellKit/ChartExport/Chart/ChartPropertiesSetup.cs:22]: Sure, I usually put lowerCase for privates... _lowerCase then?

Comment (vinkobedek) [reply Acceleratio.XCellKit/ChartExport/Chart/ChartPropertiesSetup.cs:22]: oh, sorry. wasn't paying attention. These are properties, so it's okay as is.

**Review by vinkobedek — APPROVED** [review]

File: Acceleratio.XCellKit/SpredsheetWorksheet.cs [hunk 3]
```diff
@@ -53,7 +53,7 @@ public void AddTable(SpredsheetTable table, int columnIndex, int rowIndex)
                 headerRow.AddCell(headerCell);
                 if (_maxNumberOfCharsPerColumn.Any())
                 {
-                    trackMaxChars(columnIndex + i, headerCell);
+                    trackMaxChars(columnIndex + i, headerCell, 5);
```
Comment (sglogovic) [inline Acceleratio.XCellKit/SpredsheetWorksheet.cs:56]: Filter/Sort button was over title so I added few extra spaces.

File: Acceleratio.XCellKit/SpredsheetWorksheet.cs [hunk 4]
```diff
@@ -72,10 +72,10 @@ public void AddChart(SpredsheetChart chart, int columnIndex, int rowIndex)
             _charts[new SpredsheetLocation(rowIndex, columnIndex)] = chart;
         }
 
-        private void trackMaxChars(int columnIndex, SpredsheetCell cell)
+        private void trackMaxChars(int columnIndex, SpredsheetCell cell, int extraSpace = 1)
         {
             var previousMax = _maxNumberOfCharsPerColumn[columnIndex];
-            var charsCount = cell.Value.ToString().Count() + cell.Indent; 
+            var charsCount = cell.Value.ToString().Count() + cell.Indent + extraSpace;
```
Comment (sglogovic) [inline Acceleratio.XCellKit/SpredsheetWorksheet.cs:78]: Added 1 space to make sure that last character was not always hidden.

**Review by hbagarich — APPROVED** [review]

**Pull request merged**